chai_lab/data/parsing/msas/aligned_pqt.py

```diff
@@ -49,17 +48,22 @@ def hash_sequence(seq: str) -> str:
     return hash_object.hexdigest()
 
 
+@lru_cache(maxsize=1_000_000)
+def stable_hash_for_pairkey(str) -> int:
+    # very basic, fast hash, converts to int
+    return int(hashlib.sha256(str.encode("utf-8")).hexdigest()[:7], 16)
+
+
 def expected_basename(query_sequence: str) -> str:
     """Get the expected filename based on the uppercased query sequence."""
     seqhash = hash_sequence(query_sequence.upper())
     return f"{seqhash}.aligned.pqt"
 
 
-def parse_aligned_pqt_to_msa_set(
+def parse_aligned_pqt_to_serialized_msa(
```

Comment:
We should probably rename this function to better reflect that it returns a `MSAContext` and not a "serialized MSA"